SPARQLWrapper/SmartWrapper.py

```diff
@@ -110,40 +111,37 @@ class Bindings(object):
     :vartype askResult: bool
     """
 
-    def __init__(self, retval):
+    def __init__(self, retval: QueryResult):
         """
         :param retval: the query result.
         :type retval: :class:`QueryResult<SPARQLWrapper.Wrapper.QueryResult>`
         """
         self.fullResult = retval._convertJSON()
         self.head = self.fullResult["head"]
-        self.variables = None
+        self.variables: Optional[List[str]]
         try:
             self.variables = self.fullResult["head"]["vars"]
-        except:
-            pass
+        except KeyError:
+            self.variables = None
 
-        self.bindings = []
-        try:
+        self.bindings: List[Dict[str, Value]] = []
+        if self.variables is not None:
             for b in self.fullResult["results"]["bindings"]:
                 # This is a single binding. It is a dictionary per variable; each value is a dictionary again
                 # that has to be converted into a Value instance
-                newBind = {}
+                newBind: Dict[Any, Value] = {}
                 for key in self.variables:
                     if key in b:
                         # there is a real binding for this key
                         newBind[key] = Value(key, b[key])
                 self.bindings.append(newBind)
-        except:
-            pass
 
-        self.askResult = False
         try:
             self.askResult = self.fullResult["boolean"]
-        except:
-            pass
+        except KeyError:
```

Comment:
While this makes sense to me, ideally it would go in another PR as it is not needed for type checking to pass. See [[ref](https://github.com/RDFLib/sparqlwrapper/pull/197/files#r810545424)] for discussion.